src/app/file-manager/services/limit-transferables.service.ts

```diff
@@ -1,96 +1,190 @@
-import { Injectable } from '@angular/core';
-import * as Transferables from '../actions/transferables.actions';
-import { Item } from '../models/item';
+import { Injectable, OnInit, OnDestroy } from '@angular/core';
+
+import * as Transferables from '@app/file-manager/actions/transferables.actions';
+import * as downloadActions from '@app/file-manager/actions/download-item.actions';
+import * as uploadActions from '@app/file-manager/actions/upload-item.actions';
+import * as exportToGCSActions from '@app/file-manager/actions/export-to-gcs-item.actions';
+import * as exportToS3Actions from '@app/file-manager/actions/export-to-s3-item.actions';
+
+import { Item } from '@app/file-manager/models/item';
+import { DownloadItem } from '@app/file-manager/models/download-item';
+import { UploadItem } from '@app/file-manager/models/upload-item';
+import { ExportToGCSItem } from '@app/file-manager/models/export-to-gcs-item';
+import { ExportToS3Item } from '@app/file-manager/models/export-to-s3-item';
+
 import { Store } from '@ngrx/store';
-import { AppState } from '../dbstate/app-state';
-import { FilesDatabase } from '../dbstate/files-database';
+import { AppState } from '@app/file-manager/reducers';
+
 import { ItemStatus } from '@app/file-manager/models/item-status';
+import { EntityStatus } from '@app/file-manager/models/entity-status';
 import { GcsService } from '@app/file-manager/services/gcs.service';
 import { Type } from '@app/file-manager/models/type';
 import { environment } from '@env/environment';
 import { S3ExportService } from '@app/file-manager/services/s3-export.service';
 
+import { Observable } from 'rxjs/Observable';
+import { ISubscription } from 'rxjs/Subscription';
+
+import { DownloadState } from '@app/file-manager/reducers/downloads.reducer';
+import { UploadState } from '@app/file-manager/reducers/uploads.reducer';
+import { ExportToGCSState } from '@app/file-manager/reducers/export-to-gcs.reducer';
+import { ExportToS3State } from '@app/file-manager/reducers/export-to-s3.reducer';
+
 @Injectable()
-export class LimitTransferablesService {
+export class LimitTransferablesService implements OnInit, OnDestroy {
+
+  downloadsPendingCount = 0;
+  downloadsInProgressCount = 0;
+  downloadPendingItems: { [id: string]: DownloadItem };
+
+  downSubscription: ISubscription;
+  upSubscription: ISubscription;
+  gcsSubscription: ISubscription;
+  s3Subscription: ISubscription;
+
+  downPendingCount = 0;
+  downInProgressCount = 0;
+  downPendingItems: { [id: string]: DownloadItem };
+
+  upPendingCount = 0;
+  upInProgressCount = 0;
+  upPendingItems: { [id: string]: UploadItem };
+
+  gcsPendingCount = 0;
+  gcsInProgressCount = 0;
+  gcsPendingItems: { [id: string]: ExportToGCSItem };
+
+  s3PendingCount = 0;
+  s3InProgressCount = 0;
+  s3PendingItems: { [id: string]: ExportToS3Item };
```

Comment:
This should be modeled using a class that contains the four items for each of the cases. Type generics can be used for the pendingItems field since it will be different for each case.

Reply:
I see your point. Could we put in technical debt and create a story to handle it ?. I think it is the only remaining point to fix, right ?

Reply:
OK to create a tech debt issue. I think that's all the things I found, I'll approve the PR